test/test_torch.py

```diff
@@ -17938,24 +17938,16 @@ def test_strided_mm_bmm(self, device, dtype):
 
     @skipCUDAIf(torch.version.cuda == "10.1", "flaky on CUDA 10.1")
     @onlyOnCPUAndCUDA
-    @dtypes(*torch.testing.get_all_fp_dtypes(), *torch.testing.get_all_complex_dtypes())
+    @dtypesIfCUDA(*(torch.testing.get_all_fp_dtypes(include_half=True, include_bfloat16=AMPERE_OR_ROCM) +
```

Comment:
Please don't do so. We test on all dtypes on purpose to make sure that all dtypes are tested: if it is supported, then it should run well. If it is not supported, it should raise an error.

Reply:
cc: @ngimel

Reply:
synced offline: the cpu  bmm and baddbmm has multiple code paths, some of them supports bfloat16 and float16, some don't. So depending on the input, half and bfloat could or could not be supported. https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/LinearAlgebra.cpp#L498 

So @zasdfgbnm , @ngimel and I agreed to add full support for `torch.float16` and `torch.bfloat16` in a follow-up PR and leave this one as is.